### PR TITLE
Simpify default.nix and update streamly to the latest master

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: src/Streamly/Internal/System/Process.hs [hunk 1]
```diff
@@ -93,18 +94,19 @@ module Streamly.Internal.System.Process
...
-import Control.Concurrent (forkIO)
 import Data.Function ((&))
 import Data.Word (Word8)
 import Foreign.C.Error (Errno(..), ePIPE)
 import GHC.IO.Exception (IOException(..), IOErrorType(..))
 import Streamly.Data.Array.Unboxed (Array)
 import Streamly.Data.Fold (Fold)
-import Streamly.Prelude (MonadAsync, parallel, IsStream, adapt, SerialT)
+import Streamly.Data.Stream (Stream)
+import Streamly.Prelude (MonadAsync, parallel)
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply src/Streamly/Internal/System/Process.hs:109]: See #57

File: src/Streamly/Internal/System/Process.hs [hunk 2]
```diff
@@ -357,25 +358,25 @@ createProc' modCfg path args = do
...
+    Handle -> Stream m (Array Word8) -> Stream m a
 putChunksClose h input =
     Stream.before
-        (Handle.putChunks h (adapt input) >> liftIO (hClose h))
+        (Handle.putChunks h input >> liftIO (hClose h))
         Stream.nil
 
 {-# INLINE toChunksClose #-}
-toChunksClose :: (IsStream t, MonadAsync m) => Handle -> t m (Array Word8)
+toChunksClose :: MonadAsync m => Handle -> Stream m (Array Word8)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/System/Process.hs:369]: Maybe we can do with a simpler constraint here, without MonadThrow? Does `after` require a MonadThrow?

**Comment by harendra-kumar** [comment]
doctest needs to be fixed:

```
[   0.56686] docspec.phase2: Streamly.Internal.System.Process
[   1.17719] error: toNull path args = toChunks path args & Stream.drain
[   1.17754] error: in comment at 757:1
expected: 
 but got: 
          ^
          <interactive>:513:41: error:
              Not in scope: ‘Stream.drain’
              Perhaps you meant one of these:
                ‘Stream.drop’ (imported from Streamly.Data.Stream),
                ‘Stream.drop’ (imported from Streamly.Internal.Data.Stream),
                ‘Stream.scan’ (imported from Streamly.Data.Stream)
```

**Comment by harendra-kumar** [comment]
Looks like benchmark and test do not require the `streamly` package:

```
Linking /home/runner/work/streamly-process/streamly-process/.packcheck/streamly-process-0.2.0.1/dist-newstyle/build/x86_64-linux/ghc-8.10.7/streamly-process-0.2.0.1/b/Benchmark.System.Process/build/Benchmark.System.Process/Benchmark.System.Process ...

<no location info>: error: [-Wunused-packages, -Werror=unused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - streamly-0.9.0
Linking /home/runner/work/streamly-process/streamly-process/.packcheck/streamly-process-0.2.0.1/dist-newstyle/build/x86_64-linux/ghc-8.10.7/streamly-process-0.2.0.1/t/Test.System.Process/build/Test.System.Process/Test.System.Process ...

<no location info>: error: [-Wunused-packages, -Werror=unused-packages]
    The following packages were specified via -package or -package-id flags,
    but were not needed for compilation:
      - streamly-0.9.0
cabal: Failed to build bench:Benchmark.System.Process from
streamly-process-0.2.0.1.
Failed to build test:Test.System.Process from streamly-process-0.2.0.1.
```

File: src/Streamly/Internal/System/Process.hs (hunk 2 above)
Comment (adithyaov) [reply src/Streamly/Internal/System/Process.hs:369]: Yes, `after` requires `MonadAsync`.

File: src/Streamly/Internal/System/Command.hs [hunk 3]
```diff
@@ -101,15 +105,15 @@ quotedWord =
 -- /Internal/
 {-# INLINE streamWith #-}
 streamWith :: MonadCatch m =>
-    (FilePath -> [String] -> SerialT m a) -> String -> SerialT m a
+    (FilePath -> [String] -> Stream m a) -> String -> Stream m a
 streamWith f cmd =
-    Stream.concatM $ do
-        xs <- Stream.toList
+    Stream.concatMapM (\() -> do
```
Comment (adithyaov) [inline src/Streamly/Internal/System/Command.hs:110]: This was a little weird to write. `concatM`.

File: src/Streamly/Internal/System/Command.hs [hunk 4]
```diff
@@ -63,12 +63,16 @@ import Data.Char (isSpace)
...
-import Streamly.Prelude (MonadAsync, SerialT)
 
-import qualified Streamly.Internal.Data.Fold as Fold
-import qualified Streamly.Internal.Data.Parser as Parser
-import qualified Streamly.Internal.Data.Stream.IsStream as Stream
+import qualified Streamly.Data.Fold as Fold
+import qualified Streamly.Data.Parser as Parser
+
+import qualified Streamly.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Stream as Stream (parseMany)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/System/Command.hs:74]: Why isn't parseMany exported from exposed modules? We should export it if it is not already.

File: src/Streamly/Internal/System/Command.hs (hunk 3 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/System/Command.hs:110]: We should have it in Interal.Data.Stream. And maybe should call it concatEffect.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**